react-cardknox-ifields/src/apple-pay.js

```diff
@@ -0,0 +1,477 @@
+                if (!canMakePayments) {
+                    throw new Error("Apple Pay not supported");
+                }
+            } 
+        } catch (err) {
+            lib.logError("Apple Pay initialization failed", err);
+        }
+    };
+
+    isFeatureRequested = (...features) => {
+        const {requiredFeatures} = this.state;
+        return features.some(feature => requiredFeatures.includes(feature));
+    };
+
+    validateFeatures = (obj) => {
+        if (!obj) return;
+        if (obj.error && !this.isFeatureRequested(APRequiredFeatures.address_validation)) 
+            throw new Error("Error. A Required Feature 'address_validation' must be set. Please refer to documentation on how to set 'requiredFeatures' for Address Validation");
+
+        if (obj.lineItems && 
+            !this.isFeatureRequested(APRequiredFeatures.support_recurring, APRequiredFeatures.support_subscription) &&
+            obj.lineItems.some(item =>  item.paymentTiming || 
+                                    item.recurringPaymentStartDate ||
+                                    item.recurringPaymentIntervalUnit ||
+                                    item.recurringPaymentIntervalCount ||
+                                    item.deferredPaymentDate ||
+                                    item.automaticReloadPaymentThresholdAmount)) 
+            throw new Error("Error. A Required Feature 'support_recurring' or 'support_subscription' must be set. Please refer to documentation on how to set 'requiredFeatures' for Recurring/Subscription");
+    };
+
+    getApplePayError = (error) => {
+        if (error) {
+            if (!error.code) {
+                throw new Error("'error.code' is required. For available codes refer to 'APErrorCode'");
+            }
+            lib.logError("ApplePayError custom error", error);
+            return new window.ApplePayError(error.code, error.contactField, error.message);
+        }
+        return null;
+    };
+
+    onCancel = (e) => {
+        lib.logError("Apple Pay Session canceled", JSON.stringify(e));
+        if (this.props.onCancel) {
+            this.props.onCancel(e);
+        }
+    }
+
+    onValidateMerchant = async (event) => {
+        const {session} = this.state;
+        try {
+            if (!event.isTrusted) {
+                lib.logError("onValidateMerchant", "Not trusted");
+                session.abort();
+            }
+            const response = this.props.onValidateMerchant ?
+                                await this.props.onValidateMerchant() :
+                                await this.validateMerchant();
+            const resp = typeof(response) === "string" ? JSON.parse(response) : response;
+            session.completeMerchantValidation(resp);
+        } catch (err) {
+            lib.logError("onValidateMerchant error", err);
+            session.abort();
+        }
+    }
+
+    validateMerchant = async () => {
```

Comment:
Is the httpService here just for this call?
Are you able to use axios instead?
(Same for test app which already imports axios)

Reply:
Discussed